sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -521,14 +545,22 @@ def make_predictor(self, num_thresholds):
         A TreePredictor object.
         """
         predictor_nodes = np.zeros(self.n_nodes, dtype=PREDICTOR_RECORD_DTYPE)
-        _fill_predictor_node_array(predictor_nodes, self.root,
-                                   num_thresholds, self.n_bins_non_missing)
-        return TreePredictor(predictor_nodes)
-
-
-def _fill_predictor_node_array(predictor_nodes, grower_node,
-                               num_thresholds, n_bins_non_missing,
-                               next_free_idx=0):
+        binned_categorical_bitsets = np.zeros((self.n_categorical, 8),
+                                              dtype=X_BITSET_INNER_DTYPE)
+        raw_categorical_bitsets = np.zeros((self.n_categorical, 8),
+                                           dtype=X_BITSET_INNER_DTYPE)
+        _fill_predictor_node_array(predictor_nodes, binned_categorical_bitsets,
+                                   raw_categorical_bitsets,
+                                   self.root, num_thresholds,
```

Comment:
these are not `num_thresholds` anymore, since they also refer to the output of `_find_bin_categories()`.

Well, if we agree to rely on `_find_binning_threshold()` we could keep the `num_thresholds` name... I wonder if `bin_boundaries` or `bin_limits` wouldn't be an overall better name for these things